build_tools/hack/build_env_diag.py

```diff
@@ -18,1123 +19,2202 @@
-if path:
-    (
-        PYTHON_INSTALL_DIR,
-        PYTHON_MAJOR_VERSION,
-        PYTHON_MINOR_VERSION,
-        PYTHON_PATCH_VERSION,
-    ) = where(program)
-    if {PYTHON_MAJOR_VERSION} == 2:
-        print(
-            f"{Emoji_Error} [{program}] Found Python {PYTHON_INSTALL_DIR} is Python 2."
-        )
-        printc(
-            f"""
-    TheRock Project is not supported by using Python 2.
-    Please use Python 3 version that is supported in bug-fix/security-fix status; We recommends Python 3 version >= 3.9.
+        else:
+            _stat = self.msg_stat("pass", "Disk Status", f"Disk space check pass.")
+            _except = ""
+            _result = True
+
+        return _stat, _except, _result
+
+    # ===========   General Tools/Utilities  ===========
+    def check_py(self):
+        py = self.device.python
+        py.description = "Python"
+        _env_ = py.env
+
+        if py.MAJOR_VERSION == 2:
+            _stat = self.msg_stat("err", py, f"Found Python is Python 2 at {py.exe}")
+            _except = cstring(
+                f"""
+    Found {py.name} is Python 2.
+    TheRock not support build on Python 2 environment. Please Switch to Python 3 environment.
+    We recommends you use Python 3.9 and newer versions.
+        traceback: Python major version too old
+            > expected version: 3.9.X ≤ python ≤ 3.13.X, found {py.version}
     """,
-            Color_Error,
-        )
-    if ({PYTHON_MAJOR_VERSION} == 3) and ({PYTHON_MINOR_VERSION} < 8):
-        PYTHON_VERSION_NAME = (
-            f"{PYTHON_MAJOR_VERSION}.{PYTHON_MINOR_VERSION}.{PYTHON_PATCH_VERSION}"
-        )
-        print(
-            f"{Emoji_Warning} [{program}] Found Python {PYTHON_INSTALL_DIR} is version {warn(PYTHON_VERSION_NAME)}."
-        )
-        printc(
-            f"""
-    We found your {program} version is {PYTHON_MAJOR_VERSION}.{PYTHON_MINOR_VERSION}.{PYTHON_PATCH_VERSION} <3.8.
-    We also used uv, which uv not support your version.
-    Through {PYTHON_MAJOR_VERSION}.{PYTHON_MINOR_VERSION}.{PYTHON_PATCH_VERSION} version may usable, but we don't gurantee this version is stable for TheRock build and PyTorch.
-    Please use Python 3 version that is supported in bug-fix/security-fix status; We recommends Python 3 version >= 3.9.
+                "err",
+            )
+            _result = None
+
+        elif py.MINOR_VERSION <= 8:
+            _stat = self.msg_stat(
+                "warn", py, f"Found Python {py.version} at {py.exe} {_env_}."
+            )
+            _except = cstring(
+                f"""
+    Found {py.name} version: {py.version}
+    TheRock team still not examine on these older Python versions yet, and seems will be deprecated in future release.
+    The build maybe success, but we do not promise the stability on older versions.
+        traceback: Python3 version may unstable due to Python3 version too old.
+            > expected version: 3.9.X ≤ python ≤ 3.13.X, found {py.version}
```

Comment:
Minimum supported version is Python 3.9: https://github.com/ROCm/TheRock/blob/9b873dd83d594a6abaa8db3890c09ab445651df0/CMakeLists.txt#L304-L305

We only build packages for Python 3.11+ though.

Older versions are not "deprecated" - we've never supported them and will not change that (the world has moved on from Python 3.8, and it will move on from 3.9, 3.10, etc.)